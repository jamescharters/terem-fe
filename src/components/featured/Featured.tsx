import React from "react";
import Styles from "./Featured.module.scss";
import { FeaturedItem } from "./FeaturedItem";
import { TileItem } from "@/types/TileItem";

export type FeaturedProps = {
    items: TileItem[];
};

export const Featured: React.FC<FeaturedProps> = (props) => {
    return (
        <section className={Styles.container}>
            <h3 data-testid="heading-featured-items">Featured</h3>
            <div className={Styles.items}>
                {props.items.length === 0 ? (
                    <p>Sorry, there are no featured items available.</p>
                ) : (
                    props.items.map((item) => <FeaturedItem key={item.title} {...item} />)
                )}
            </div>
        </section>
    );
};

export default Featured;
