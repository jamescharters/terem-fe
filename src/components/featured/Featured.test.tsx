import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { Featured } from "./Featured";

describe("Featured", () => {
    it("renders correctly", () => {
        const mockItems = [
            {
                imageUrl: "http://a.b",
                linkUrl: "http://b.c",
                location: "Sydney, Australia",
                title: "A Fake Title",
            },
        ];
        const result = render(<Featured items={mockItems} />);
        expect(result.asFragment()).toMatchSnapshot();
    });

    it("renders correctly if there are no items", () => {
        const result = render(<Featured items={[]} />);
        expect(result.asFragment()).toMatchSnapshot();
    });
});
