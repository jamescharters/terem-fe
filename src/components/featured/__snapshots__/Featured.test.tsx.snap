// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Featured > renders correctly 1`] = `
<DocumentFragment>
  <section
    class="_container_7e7544"
  >
    <h3>
      Featured
    </h3>
    <div
      class="_items_7e7544"
    >
      <div
        class="_container_740b7c"
        style="background-image: url(http://a.b);"
      >
        <div>
          <h4>
            A Fake Title
          </h4>
          <p>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-location-pin fa-1x "
              data-icon="location-pin"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
                fill="currentColor"
              />
            </svg>
            <span>
              Sydney, Australia
            </span>
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Featured > renders correctly if there are no items 1`] = `
<DocumentFragment>
  <section
    class="_container_7e7544"
  >
    <h3>
      Featured
    </h3>
    <div
      class="_items_7e7544"
    >
      <p>
        Sorry, there are no featured items available.
      </p>
    </div>
  </section>
</DocumentFragment>
`;
